src/ExpressionEvaluator/VisualBasic/Source/ExpressionCompiler/CompilationContext.vb

```diff
@@ -1287,6 +1287,11 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.ExpressionEvaluator
                     ' locals above.  We assert that we are seeing the latter.
                     Debug.Assert((variableKind = DisplayClassVariableKind.Parameter) OrElse
                         (variableKind = DisplayClassVariableKind.Me))
+
+                    If variableKind = DisplayClassVariableKind.Parameter AndAlso instance.Type.Name.StartsWith(StringConstants.DisplayClassPrefix, StringComparison.Ordinal) Then
```

Comment:
GeneratedNames.GetKind does not include a LambdaDisplayClass so i have made that check in the if statement, maybe LambdaDisplayClass should be added to the GeneratedNameKind enum.